src/main/java/mekanism/common/content/gear/mekatool/ModuleBlastingUnit.java

```diff
@@ -0,0 +1,110 @@
+import mekanism.api.gear.config.ModuleEnumData;
+import mekanism.api.text.EnumColor;
+import mekanism.api.text.IHasTextComponent;
+import mekanism.api.text.TextComponentUtil;
+import mekanism.common.MekanismLang;
+import mekanism.common.block.BlockBounding;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+
+@ParametersAreNonnullByDefault
+public class ModuleBlastingUnit implements ICustomModule<ModuleBlastingUnit> {
+
+    private IModuleConfigItem<BlastRadius> blastRadius;
+
+    @Override
+    public void init(IModule<ModuleBlastingUnit> module, ModuleConfigItemCreator configItemCreator) {
+        blastRadius = configItemCreator.createConfigItem("blast_radius", MekanismLang.MODULE_BLAST_RADIUS,
+                new ModuleEnumData<>(BlastRadius.class, module.getInstalledCount() + 1, BlastRadius.LOW));
+    }
+
+    public int getBlastRadius() {
+        return blastRadius.get().getRadius();
+    }
+
+    public static Set<BlockPos> findPositions(Level world, BlockPos location, Direction axis, int radius) {
+        int minX = 0, minY = 0, minZ = 0;
+        int maxX = 0, maxY = 0, maxZ = 0;
+        switch (axis) {
+            case UP, DOWN -> {
+                minX = -radius;
+                maxX = radius;
+                minY = maxY = 0;
+                minZ = -radius;
+                maxZ = radius;
+            }
+            case EAST, WEST -> {
+                minX = maxX = 0;
+                minY = (radius == 0) ? 0 : -1;
+                maxY = (radius == 0) ? 0 : (2 * radius - 1);
+                minZ = -radius;
+                maxZ = radius;
+            }
+            case NORTH, SOUTH -> {
+                minX = -radius;
+                maxX = radius;
+                minY = (radius == 0) ? 0 : -1;
+                maxY = (radius == 0) ? 0 : (2 * radius - 1);
+                minZ = maxZ = 0;
+            }
+        }
+        Set<BlockPos> found = new HashSet<>();
+        for (BlockPos pos : BlockPos.betweenClosed(location.offset(minX, minY, minZ), location.offset(maxX, maxY, maxZ))) {
+            BlockState blockState = world.getBlockState(pos);
+            if (!blockState.isAir() && !(blockState.getBlock() instanceof BlockBounding) && !(blockState.destroySpeed < 0)) {
+                found.add(pos.immutable());
+            }
+        }
+        return found;
+    }
+
+    @Override
+    public void addHUDStrings(IModule<ModuleBlastingUnit> module, Player player, Consumer<Component> hudStringAdder) {
+        //Only add hud string if enabled in config
+        if (module.isEnabled()) {
+            hudStringAdder.accept(MekanismLang.MODULE_BLAST_RADIUS.translateColored(EnumColor.DARK_GRAY, EnumColor.INDIGO, blastRadius.get().getTextComponent()));
```

Comment:
Manual call to `getTextComponent` should not be needed here as `ILangEntry`'s `translate` methods go through our `TextComponentUtil` that has special handling for objects that are `instanceof IHasTextComponent`